MekHQ/src/mekhq/gui/dialog/ImageChoiceDialog.java

```diff
@@ -74,125 +76,100 @@
+                             boolean force) {
         super(parent, modal);
         this.category = category;
-        filename = file;
+        this.filename = filename;
         imagesMouseAdapter = new ImageTableMouseAdapter();
         this.imageItems = items;
         this.force = force;
-        // If we're doing forces, initialize the hashmap for use
-        if (force) {
-            iconMap = new LinkedHashMap<>();
-        }
-        initComponents();
-        fillTable((String) comboCategories.getSelectedItem());
-        int rowIndex = 0;
-        for(int i = 0; i < imageTableModel.getRowCount(); i++) {
-            if (imageTableModel.getValueAt(i, 0).equals(filename)) {
-                rowIndex = i;
-                break;
+
+        // Clone the input iconMap
+        this.iconMap = new LinkedHashMap<>();
+        if ((iconMap != null) && !iconMap.isEmpty()) {
+            for (Map.Entry<String, Vector<String>> entry : iconMap.entrySet()) {
+                if ((entry.getValue() != null) && !entry.getValue().isEmpty()) {
+                    this.iconMap.put(entry.getKey(), new Vector<>(entry.getValue()));
+                }
             }
         }
-        tableImages.setRowSelectionInterval(rowIndex, rowIndex);
+
+        initComponents();
+
         setLocationRelativeTo(parent);
         setUserPreferences();
     }
+    //endregion Constructors
 
     private void initComponents() {
+        ResourceBundle resourceMap = ResourceBundle.getBundle("mekhq.resources.ImageChoiceDialog", new EncodeControl()); //$NON-NLS-1$
         GridBagConstraints gbc;
 
-        scrImages = new JScrollPane();
-        tableImages = new JTable();
-        comboCategories = new JComboBox<>();
-        btnSelect = new JButton();
-        btnCancel = new JButton();
-        JPanel imagesPanel = new JPanel();
         getContentPane().setLayout(new GridBagLayout());
-
-        ResourceBundle resourceMap = ResourceBundle.getBundle("mekhq.resources.ImageChoiceDialog", new EncodeControl()); //$NON-NLS-1$
         setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
         setName("Form"); // NOI18N
         setTitle(force ? resourceMap.getString("Force.title") : resourceMap.getString("Portrait.title"));
+
+        JPanel imagesPanel = new JPanel();
         imagesPanel.setLayout(new GridBagLayout());
         imagesPanel.setName(PANEL_IMAGES);
 
-        scrImages.setName("scrImages"); // NOI18N
+        DefaultComboBoxModel<String> categoryModel = new DefaultComboBoxModel<>();
+        String match = null;
+        categoryModel.addElement(Crew.ROOT_PORTRAIT);
+        Iterator<String> names = (imageItems != null)
+                ? imageItems.getCategoryNames() : Collections.emptyIterator();
+        while (names.hasNext()) {
+            String name = names.next();
+            if (!"".equals(name)) {
+                categoryModel.addElement(name);
+                if (category.equals(name)) {
+                    match = name;
```

Comment:
Should you break here?

Reply:
It can only match once, so it definitely should.  Fixed.